src/services/item/controller.ts

```diff
@@ -46,11 +51,19 @@ const plugin: FastifyPluginAsync = async (fastify) => {
         query: { parentId },
         body: data,
       } = request;
+
+      // lat and lng should exist together
+      const { lat, lng } = data;
+      if ((lat && !lng) || (lng && !lat)) {
+        throw new PartialItemGeolocation({ lat, lng });
+      }
+
       return await db.transaction(async (manager) => {
         const repositories = buildRepositories(manager);
         const item = await itemService.post(member, repositories, {
           item: data,
           parentId,
+          ...(data.lng && data.lat ? { geolocation: { lng: data.lng, lat: data.lat } } : {}),
```

Comment:
I don't need the check anymore

Reply:
You still have to check that lat and long are defined, don't you ?